Fix: Miscellanous changes to keep header and hero style consistent

aqsaaqeel opened the pull request:
## Summary

- The main hero section on the landing page was not showing the image. This was caused because there wasn't a conditional in place for mobile size.
- The `featured projects` section's heading and button on the landing page were not consistent with that of the rest of the sections.
- The navbar width was set to 100% on desktop screens which is not required according to current design.
- The avatar button was blank when not logged in.
- The hero section contents width was not aligned with navbar.

## Closes #934 

## Screenshots

Image now appearing normally:

![Screenshot from 2023-10-21 06-02-47](https://github.com/unstructuredstudio/zubhub/assets/85514520/b58dc4af-4108-4177-afcd-577d697bc1ec)

Featured Projects heading consistent now:

![Screenshot from 2023-10-21 06-04-00](https://github.com/unstructuredstudio/zubhub/assets/85514520/59dca0f0-5098-41d5-b724-a88859779cd3)

Featured Projects buttons style consistent with rest of the buttons:

![Screenshot from 2023-10-21 06-04-04](https://github.com/unstructuredstudio/zubhub/assets/85514520/426342d1-1c48-46cc-9c57-aa09392c1093)

## Concerns after this PR

The featured projects section is rendered as a whole component. This component is also used on the dashboard. Changing the heading style changes it on the dashboard too. Is this alright? 

![Screenshot from 2023-10-21 06-09-40](https://github.com/unstructuredstudio/zubhub/assets/85514520/87f3fb6d-7fb2-4233-8946-f76b6cfd6408)

Commit:
fix: hero image on mobile

Commit:
fix: Featured Projects heading and buttons size

srish commented:
Closing as a duplicate of https://github.com/unstructuredstudio/zubhub/issues/934

aqsaaqeel commented:
> Closing as a duplicate of #934

hey @srish this is a PR I raised to fix #934 🤔 why was this closed?

tuxology commented:
@aqsaaqeel Please see the updated scope on ticket #934. Update the PR description and title as well on this. Reopening this.

aqsaaqeel commented:
> @aqsaaqeel Please see the updated scope on ticket #934. Update the PR description and title as well on this. Reopening this.

Thanks, on it!

Commit:
revert: navbar width

tuxology commented:
@aqsaaqeel Are you working on this? This is important since we need to fix regressions we introduced.

Commit:
fix: added pointer cursor to icons

Commit:
fix: placeholder avatar icon when unauthenticated

Commit:
fix: hero section width consistent with navbar

Commit:
fix: avatar background color

aqsaaqeel commented:
> @aqsaaqeel Are you working on this? This is important since we need to fix regressions we introduced.

I've made the changes, my concern is the hero section layout. The new Figma design was suitable for 100% width. For now, I have added multiple breakdowns to make it responsive (so it is easier to revert any changes if required)

aqsaaqeel committed:
Merge branch 'master' into fix/934

aqsaaqeel commented:
Regression issue fixed with this PR

1. Hero section image on mobile screens

![Screenshot from 2023-10-28 12-05-29](https://github.com/unstructuredstudio/zubhub/assets/85514520/b6500626-c0ff-4e02-8fe4-ff516ac9cf3b)

2. Navbar width for its content reverted to its initial state

![Screenshot from 2023-10-28 12-08-24](https://github.com/unstructuredstudio/zubhub/assets/85514520/8adf2975-e0f8-48c5-a6f1-9a0cb5184323)

3.  Hero Section container to be consistent with navbar contents.

![Screenshot from 2023-10-28 11-31-33](https://github.com/unstructuredstudio/zubhub/assets/85514520/83c569ae-95a1-4c28-89c6-a01696bbaa0a)

4. Avatar icon to show default avatar when unauthenticated.

![Screenshot from 2023-10-28 12-08-24](https://github.com/unstructuredstudio/zubhub/assets/85514520/f7f74e8d-897b-4baf-ace3-4b6ff0161558)

when authenticated:

![Screenshot from 2023-10-28 12-11-38](https://github.com/unstructuredstudio/zubhub/assets/85514520/b278c6df-2b27-4091-840e-4b3cab79ee60)

5. Heading and buttons for featured projects to be consistent with the rest of the headings and buttons on the landing page.

![Screenshot from 2023-10-28 11-32-57](https://github.com/unstructuredstudio/zubhub/assets/85514520/311ce50f-d24d-42b7-a769-af678d9e697b)

aqsaaqeel commented:
@kamthamc @tuxology This is ready for review

kamthamc commented on zubhub_frontend/zubhub/src/assets/js/styles/components/staff_pick/staffPickStyles.js:
```diff
@@ -13,6 +13,10 @@ const styles = theme => ({
   },
+  MessagePrimaryStyle: {
+    fontSize: '2.5rem',
+    fontWeight: 1000,
```
it shouldn't be 1000 🤔

aqsaaqeel replied:
Since the aim was to make the heading for the Featured Project section the same as the headings on other sections on the landing page, I have used the same style used in those headings.

chaitanya-cloudknox replied:
Sorry my mistake was thinking of some other property which should be <10

kamthamc commented on zubhub_frontend/zubhub/src/components/avatar_button/AvatarButton.jsx:
```diff
@@ -23,25 +23,35 @@ const AvatarButton = props => {
-          src={auth?.avatar}
-        />
+        {!auth.token ? (
+          <Avatar
```
the usage looks same. may be just change props if token is available

aqsaaqeel replied:
Thank you for the review, I have refactored the code.

kamthamc commented on zubhub_frontend/zubhub/src/views/home/Home.jsx:
```diff
@@ -161,8 +161,8 @@ function Projects(props) {
-                         src={hero.image_url}
-                         alt=""
+                         src={hero.image_url ? hero.image_url : heroMainImage}
+                         alt="Six animated children happily walking"
```
it needs to be translated

Commit:
refactored Avatar component code

Commit:
fix: added translated text for image alt prop

Commit:
Merge branch 'fix/934' of ssh://github.com/aqsaaqeel/zubhub into fix/934

tuxology commented on zubhub_frontend/zubhub/src/assets/js/styles/views/projects/projectsStyles.js:
```diff
@@ -25,21 +25,32 @@ const styles = theme => ({
+    // maxWidth: '1230px',
     marginBottom: '2em',
     height:'100vh',
+    [theme.breakpoints.down('1620')]: {
```
I don't feel adding 3 extra breakpoints - two of which are very close and very arbitrary is the correct way to solve this. Breakpoints should remain fixed and within them, the hero elements can size relatively.  For example. explore use of `em` for font sizing and for padding, there is some other underlying issue that padding is not adjusting on existing breakpoints. I am sure there is a much cleaner way to do this

tuxology reviewed (changes requested):
We still need to figure out a way to address the hero element issue properly. We may not need many breakpoitns if the padding/font sizing is handled properly.  We do need this change sooner than later.

aqsaaqeel commented:
Can I deviate from the figma design a bit and test with smaller font size to accommodate the padding?

Commit:
fix: added dynamic padding to hero section

Commit:
fix: removed extra padding

aqsaaqeel commented:
@tuxology I have now used calc for taking care of padding and the layout looks responsive, please review

tuxology reviewed (approved):
OK, this LGTM for now - much cleaner as well. Lets merge and see if there are improvements 👍
